src/App.jsx

```diff
@@ -14,45 +17,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_TYPE_LENG = 'leng';
+const SORT_TYPE_ALPH = 'alph';
+
+const updateGoods = (goods, { sortType, isReverse }) => {
+  if (sortType === '' && !isReverse) {
+    return goods;
+  }
+
+  const udatedGoods = [...goods];
+
+  if (sortType === SORT_TYPE_LENG || sortType === SORT_TYPE_ALPH) {
+    udatedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SORT_TYPE_LENG:
+          return good1.length - good2.length;
+        case SORT_TYPE_ALPH:
+          return good1.localeCompare(good2);
+        default:
+          throw new Error('Unbelievable error');
+      }
+    });
+  }
+
+  if (isReverse) {
+    udatedGoods.reverse();
+  }
+
+  return udatedGoods;
+};
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+
+  const goods = updateGoods(goodsFromServer, { sortType, isReverse });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': SORT_TYPE_ALPH !== sortType,
+          })}
+          onClick={() => setSortType(SORT_TYPE_ALPH)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': SORT_TYPE_LENG !== sortType,
+          })}
+          onClick={() => setSortType(SORT_TYPE_LENG)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReverse,
+          })}
+          onClick={() => setIsReverse(!isReverse)}
+        >
+          Reverse
+        </button>
```

Comment:
In future, when you gonna use typescript, don't forget that you can iterate through 'enum' of sort type options, and render these buttons in easier way